js/tji-datasets-explore.js

```diff
@@ -220,33 +220,58 @@ TJIChartView.prototype.column_whitelist = [
 // of the charts, filter panel, etc.
 TJIChartView.prototype.get_data = function() {
   var that = this;
-  jQuery.getJSON('/cleaned_custodial_death_reports.json')
+  var url = '/cdr_compressed.json';
+  jQuery.getJSON(url)
     .done(function(data){
       that.state.data = data;
-      that.parse_data();
+      that.decompress_data();
+      that.transform_data();
       that.create_charts();
       that.create_filter_panel();
       that.attach_events();
     })
-    .fail(function(e){
-      console.log('error fetching data from: ' + this.state.url, e);
+    .fail(function(jqxhr, textStatus, error){
+      console.log('error fetching data from: ' + url, error);
     })
 }
 
+/* The data is compressed into a small json object for fast page loading,
+ * which we decompress here for convenient manipulation in this app.
+ * 
+ * Currently, the data is compressed by this script in our data-processing repo:
+ * https://github.com/texas-justice-initiative/data-processing/blob/master/data_cleaning/create_compressed_cdr_for_website.ipynb 
+ * 
+ * See that file for an explanation of the compression and examples.
+ */
+TJIChartView.prototype.decompress_data = function() {
+  // We want a list of json objects, one per record. We will build these
+  // out incrementally.
+  var new_data = [];
+  _.times(this.state.data['meta']['num_records'], function(){ new_data.push({}); });
+
+  var that = this;
+  var meta = this.state.data.meta;
+  var records = this.state.data.records;
+  _.forEach(records, function(values, column) {
```

Comment:
can we use _.each instead of _.forEach. They're aliases and I prefer to be consistent so there's less opportunity for confusion.